qiling/os/posix/syscall/poll.py

```diff
@@ -4,6 +4,34 @@
 #
 
 from qiling import Qiling
+import select
 
 def ql_syscall_poll(ql: Qiling, fds: int, nfds: int, timeout: int):
-    return 0
\ No newline at end of file
+    fn_map = {}
+    try:
+        p = select.poll()
+        for i in range(nfds):
+            fd = ql.mem.read_ptr(fds + i*8, 4)
```

Comment:
Consider using a `BaseStruct` for that, instead of doing pointer arithmetics and reading memory.
Specifically, look at the `ref` method which woud be a good fir here.